File: src/main/java/com/javarush/island/shevchenko/services/PlantGrowthTask.java
package com.javarush.island.shevchenko.services;

import com.javarush.island.shevchenko.entities.islandMap.Location;


/**
 * Класс PlantGrowthTask представляет собой задачу роста растений на определенной локации
 * на острове животных. Класс реализует интерфейс Runnable, позволяя выполнять рост растений
 * в отдельном потоке.
 */



public class PlantGrowthTask implements Runnable {
    private final Location location;
    public PlantGrowthTask(Location location) {
        this.location = location;
    }

    public void growUp() {
        location.plantGrowth();
    }

    @Override
    public void run() {
        location.getLock().lock();
        try {
            growUp();
        } finally {
            location.getLock().unlock();
        }
    }
}

File: src/main/kotlin/ru/cooper/island/services/PlantGrowthTask.kt
package ru.cooper.island.services

import ru.cooper.island.entities.islandMap.Location

/**
 * Класс PlantGrowthTask представляет собой задачу роста растений на определенной локации
 * на острове животных. Класс реализует интерфейс Runnable, позволяя выполнять рост растений
 * в отдельном потоке.
 */
class PlantGrowthTask(private val location: Location) : Runnable {
    fun growUp() {
        location.plantGrowth()
    }

    override fun run() {
        location.lock.lock()
        try {
            growUp()
        } finally {
            location.lock.unlock()
        }
    }
}
